migrations/1609942742732-fixStorageProviderInconsistencies.js

```diff
@@ -0,0 +1,166 @@
+const mongoose = require('mongoose');
+const aws = require('aws-sdk');
+const { Configuration } = require('@hpi-schul-cloud/commons');
+const CryptoJS = require('crypto-js');
+
+// eslint-disable-next-line no-unused-vars
+const { info, warning } = require('../src/logger');
+
+const { connect, close } = require('../src/utils/database');
+
+const { storageProviderSchema } = require('../src/services/storageProvider/model');
+const { schoolSchema } = require('../src/services/school/model');
+
+const StorageProviderModel = mongoose.model('storageprovider65373489214', storageProviderSchema, 'storageproviders');
+const SchoolModel = mongoose.model('schools34838583553', schoolSchema, 'schools');
+
+// How to use more than one schema per collection on mongodb
+// https://stackoverflow.com/questions/14453864/use-more-than-one-schema-per-collection-on-mongodb
+
+const EMPTY_PROVIDER = 'NOT_FOUND';
+
+const getConfig = (provider) => {
+	const awsConfig = new aws.Config({
+		signatureVersion: 'v4',
+		s3ForcePathStyle: true,
+		sslEnabled: true,
+		accessKeyId: provider.accessKeyId,
+		secretAccessKey: provider.secretAccessKey,
+		region: provider.region,
+		endpointUrl: provider.endpointUrl,
+	});
+	awsConfig.endpoint = new aws.Endpoint(provider.endpointUrl);
+	return awsConfig;
+};
+
+const getAWSS3 = async (storageProvider) => {
+	const S3_KEY = Configuration.get('S3_KEY');
+
+	storageProvider.secretAccessKey = CryptoJS.AES.decrypt(storageProvider.secretAccessKey, S3_KEY).toString(
+		CryptoJS.enc.Utf8
+	);
+	return new aws.S3(getConfig(storageProvider));
+};
+
+const getBuckets = async (provider) => {
```

Comment:
this and the next looks like a repo, extract this?
```suggestion
const getBucketNamesByProvider = async (provider) => {
```
please document input and return types everywhere.